tests/src/application/controllers/signup.spec.ts

```diff
@@ -1,295 +1,96 @@
+      expect(httpResponse).toEqual(conflict(new ConflictGuardianError()))
     })
-  })
 
-  describe('test the execution of the AddGuardian use case', () => {
-    it('Should return 500 if AddGuardian throws', async () => {
+    it('Should return 500 (ServerError) if AddGuardian throws', async () => {
       const { sut, addGuardianStub } = makeSut()
-      jest.spyOn(addGuardianStub, 'add').mockImplementationOnce(async () => {
-        return await Promise.reject(new Error())
-      })
-
-      const httpRequest = makeFakeSignUpRequest()
-
+      jest.spyOn(addGuardianStub, 'add').mockRejectedValue(new Error())
       const httpResponse = await sut.handle(httpRequest)
       expect(httpResponse).toEqual(makeFakeServerError())
     })
 
     it('Should call AddGuardian with correct values', async () => {
       const { sut, addGuardianStub } = makeSut()
       const addGuardianSpy = jest.spyOn(addGuardianStub, 'add')
-
-      const httpRequest = makeFakeSignUpRequest()
       await sut.handle(httpRequest)
-
       expect(addGuardianSpy).toHaveBeenCalledWith({
-        firstName: 'any_first_name',
-        lastName: 'any_last_name',
-        email: 'any_email@mail.com',
-        phone: 'any_phone',
-        password: 'any_password',
+        firstName: httpRequest.body.firstName,
+        lastName: httpRequest.body.lastName,
+        email: httpRequest.body.email,
+        password: httpRequest.body.password,
+        phone: httpRequest.body.phone,
         verificationToken: 'token dumb'
       })
     })
   })
 
-  describe('test signup controller success case', () => {
-    it('Should return 201 if valid data is provided', async () => {
-      const { sut } = makeSut()
-
-      const response = makeFakeGuardianData({ withId: true })
-      const httpRequest = makeFakeSignUpRequest()
+  describe('Validation', () => {
+    it('Should return 400 (BadRequest) if Validation returns an error', async () => {
+      const { sut, validationStub } = makeSut()
+      jest.spyOn(validationStub, 'validate').mockReturnValue(new MissingParamError('email'))
       const httpResponse = await sut.handle(httpRequest)
+      expect(httpResponse).toEqual(badRequest(new MissingParamError('email')))
+    })
 
-      expect(httpResponse).toEqual(create(response))
+    it('Should call Validation with correct value', async () => {
+      const { sut, validationStub } = makeSut()
+      const validateSpy = jest.spyOn(validationStub, 'validate')
+      await sut.handle(httpRequest)
+      expect(validateSpy).toHaveBeenCalledWith({
+        firstName: httpRequest.body.firstName,
+        lastName: httpRequest.body.lastName,
+        email: httpRequest.body.email,
+        phone: httpRequest.body.phone,
+        password: httpRequest.body.password,
+        passwordConfirmation: httpRequest.body.passwordConfirmation
+      })
     })
   })
+
+  test('Should return 201 (Create) if valid data are provide', async () => {
```

Comment:
Status 201 is "Created"

Reply:
yes, I agree. In the next branches I'll take this name to identify the statuses, I haven't taken it yet because it's very busy, okay?